control.lua

```diff
@@ -156,165 +158,205 @@ end
-	for _,v in pairs(VALID_POWER_TRANSFER_RATES) do
-		if v == rate then return v end
+function power_middleman_surface()
+	if game.surfaces["factory-power-connection"] then
+		return game.surfaces["factory-power-connection"]
+	end
+	
+	if #game.surfaces == 256 then
+		error("Unfortunately you have no available surfaces left for Factorissimo2. You cannot use Factorissimo2 on this map.")
 	end
-	return 0 -- Catchall
+	
+	local map_gen_settings = {height=1, width=1, property_expression_names={}}
+	map_gen_settings.autoplace_settings = {
+		["decorative"] = {treat_missing_as_default=false, settings={}},
+		["entity"] = {treat_missing_as_default=false, settings={}},
+		["tile"] = {treat_missing_as_default=false, settings={["out-of-map"]={}}},
+	}
+	
+	local surface = game.create_surface("factory-power-connection", map_gen_settings)
+	surface.set_chunk_generated_status({0, 0}, defines.chunk_generated_status.entities)
+	surface.set_chunk_generated_status({-1, 0}, defines.chunk_generated_status.entities)
+	surface.set_chunk_generated_status({0, -1}, defines.chunk_generated_status.entities)
+	surface.set_chunk_generated_status({-1, -1}, defines.chunk_generated_status.entities)
+	
+	return surface
 end
 
-local function update_power_settings(factory)
-	if factory.built then
-		local layout = factory.layout
-		-- Inside sender
-		local new_ies = factory.inside_surface.create_entity{
-			name = "factory-power-output-2-" .. factory.transfer_rate,
-			position = {factory.inside_x + layout.inside_energy_x, factory.inside_y + layout.inside_energy_y},
-			force = factory.force
-		}
-		new_ies.destructible = false
-		new_ies.operable = false
-		new_ies.rotatable = false
-		if factory.inside_energy_sender.valid then
-			factory.inside_energy_sender.destroy()
+local function remove_direct_connection(factory)
+	local dc = factory.direct_connection
+	if not dc or not dc.valid then return end
+	
+	for _, pole in pairs(factory.inside_power_poles) do
+		for _, neighbour in pairs(pole.neighbours.copper) do
+			if neighbour == dc then
+				local old = {}
+				for _, neighbour in ipairs(dc.neighbours.copper) do
+					if neighbour ~= pole then old[#old+1] = neighbour end
+				end
+				dc.disconnect_neighbour()
+				for _, neighbour in ipairs(old) do
+					dc.connect_neighbour(neighbour)
+				end
+				factory.direct_connection = nil
+				return
+			end
 		end
-		factory.inside_energy_sender = new_ies
+	end
+end
 
-		-- Inside receiver
-		local new_ier = factory.inside_surface.create_entity{
-			name = "factory-power-input-2-" .. factory.transfer_rate,
-			position = {factory.inside_x + layout.inside_energy_x, factory.inside_y + layout.inside_energy_y},
-			force = factory.force
-		}
-		new_ier.destructible = false
-		new_ier.operable = false
-		new_ier.rotatable = false
-		if factory.inside_energy_receiver.valid then
-			factory.inside_energy_receiver.destroy()
+local function delete_middleman(i)
+	local pole = global.middleman_power_poles[i]
+	if pole == 0 then return end
```

Comment:
Why is the pole entry not set to nil? setting it to 0 is strange

Reply:
This is a necessary step for this line
local middleman = surface.create_entity{name = "factory-power-connection", position = {2*(n%32), 2*math.floor(n/32)}, force = "neutral"}
The value "n" is calculated by searching for the first 0 aka gap in the pole grid.
If it was set to nil then I would not be able to find this gap.
There are other ways of solving this problem but I believe this is the cleanest